### PR TITLE
Update component callbacks

### LINKED ISSUE #645: The method used to raise events on components is inconsistent
An example of this can be seen in `NumberInput`:

```
  onChange: (event: any) => void
```

A preferred approach might be:

```
  onChange: (event: React.ChangeEvent<{}>, value: number) => void
```

It is recommended that all components are audited to ensure a consistent approach. This will be a breaking change and needs to wait for `v3`.

### PR DESCRIPTION
**Pull request by thyhjwb6**
## Related issue
Closes #645 

## Overview
Updates component callbacks so that the event is the first parameter.

## Reason
Consistency with other callbacks and standard practice.

## Work carried out
- [x] Update callbacks

### CODE REVIEW

File: packages/react-component-library/src/components/CheckboxEnhanced/CheckboxEnhanced.tsx [hunk 1]
```diff
@@ -25,7 +25,6 @@ export const CheckboxEnhanced: React.FC<CheckboxEnhancedProps> = ({
   name,
   tabIndex = 0,
   title,
-  isInvalid,
```
Comment (jpveooys) [inline packages/react-component-library/src/components/CheckboxEnhanced/CheckboxEnhanced.tsx]: Hmm seems like `isInvalid` was/is broken here... but I guess CheckboxEnhanced doesn't matter for v3?

File: packages/react-component-library/src/components/Pagination/usePageChange.tsx [hunk 2]
```diff
@@ -1,24 +1,55 @@
...
+  )
+
+  const onPaginationButtonClickHandler = useCallback(
+    (pageNumber: number) => (event: React.MouseEvent) => {
+      changePage(event, pageNumber)
+    },
+    [changePage]
+  )
+
+  // const onPaginationButtonClickHandler = useCallback(
```
Comment (jpveooys) [inline packages/react-component-library/src/components/Pagination/usePageChange.tsx]: Did you mean to leave this in?

File: packages/react-component-library/src/components/TabSet/TabSet.tsx [hunk 3]
```diff
@@ -163,12 +170,7 @@ export const TabSet: React.FC<TabSetProps | ScrollableTabSetProps> = ({
         {Children.map(
           children,
           (child: React.ReactElement<TabProps>, index: number) => {
-            const {
-              children: tabChildren,
-              title,
-              isActive,
-              ...tabRest
-            } = child.props
+            const { children: tabChildren, title, ...tabRest } = child.props
```
Comment (jpveooys) [inline packages/react-component-library/src/components/TabSet/TabSet.tsx]: Think we need to destructure `isActive` here to drop it from `tabRest` (as its set below). (Think this is the conflict as well...)

Comment (thyhjwb6) [reply packages/react-component-library/src/components/TabSet/TabSet.tsx]: I rebased so this commit should be updated.

**Review by jpveooys — APPROVED** [review]

**Review by m7kvqbe1 — APPROVED** [review]
Nice change!

**Pull request merged**